api_app/api/routes/workspaces.py

```diff
@@ -147,6 +164,16 @@ async def delete_workspace(response: Response, workspace=Depends(get_workspace_b
     return OperationInResponse(operation=operation)
 
 
+@workspaces_core_router.post("/workspaces/{workspace_id}/invoke-action", status_code=status.HTTP_202_ACCEPTED, response_model=OperationInResponse, name=strings.API_INVOKE_ACTION_ON_WORKSPACE)
+async def invoke_action_on_workspace(response: Response, action: str, workspace=Depends(get_workspace_by_id_from_path), resource_template_repo=Depends(get_repository(ResourceTemplateRepository)), operations_repo=Depends(get_repository(OperationRepository)), user=Depends(get_current_workspace_owner_or_researcher_user)) -> OperationInResponse:
+    validate_user_is_workspace_owner_or_resource_owner(user, workspace)
```

Comment:
is this needed given the `Depends` on 168?

Reply:
Right you are sir! Thanks

Reply:
Fixed